### DISCUSSION TITLE
Paging with FetchPatientRecordAsync

### DISCUSSION THREAD

**Discussion by johnkwaters** [FirelyTeam/firely-net-sdk#1982] [Q&A] [answered]
I found that when using FetchPatientRecordAsync, not all records are being returned, and this is because if you do the same $everything call manually, it returns more than one page,

Is there a way increase the count parameter on the query that gets issued, for instance

GET https://fhir-st-mpi-clinical-dev.apps.okd.healthconcourse.com/Patient/91b4eb38-d1a0-453f-9cb0-b5de3a01b1bd/$everything?_count=1000

**Comment by mbaltus** [comment, marked as answer]
I don't think the FetchPatientRecord method has an overload to specify the _count. If you need that, you would need to set up the call using the more generic InstanceOperation and a Parameters resource:

``` csharp
  var input = new Parameters();
  input.Add("_count", new Integer(1000));
  var patientRecord = c.InstanceOperation(new ResourceIdentity("Patient/[id]"), "everything", input, useGet:true);
```

That said, the server may not allow a client to increase the count, so you'll have to test this.
Once you have the result Bundle, you can use the FhirClient.Continue method to page to the next(/previous/first/last) page.

**Reply by johnkwaters** [reply]
Thanks. I feel bad asking all these questions, but the help referenced in GitHub only contains the bare basics.  What is a good place to go to learn about all these details and nuances?

**Reply by johnkwaters** [reply]
And how to I convert the result of InstanceOperation to a Patient resource?

**Reply by mbaltus** [reply]
There is a bit more documentation here: https://docs.fire.ly/projects/Firely-NET-SDK/. Also, looking at the unit tests in the SDK source could be insightful and has often helped me to get an idea of what methods can be used.

The result of the operation is of type Resource. In this case you know it will be a Bundle if the operation was successful, since that is defined as outcome for $everything. You can simply cast the result to a Bundle - to be sure with an extra test checking the resource type - and then the Patient should be in the first entry of that Bundle. You can also use the methods on the entry list to find the Patient, if it is not in the first entry.
